Upgrade dependencies

Using an up-to-date eslint will solve some of my problems.

I would do this myself, however I'm running into issues I can't solve since I'm not familiar with writing either eslint or broccoli plugins.

First of all, eslint no longer lints the project. I'm not sure if it ever did?

When I enable options that allow it to lint the project again, I run into this issue:

```
-> % npm test

> broccoli-lint-eslint@1.1.1 test /Users/devinus/Projects/broccoli-lint-eslint
> rm -rf build/ && npm run lint && broccoli build build > broccoli-build.out && node node_modules/mocha/bin/mocha build/test.js


> broccoli-lint-eslint@1.1.1 lint /Users/devinus/Projects/broccoli-lint-eslint
> eslint .

Error: Plugin subclasses must call the superclass constructor: Plugin.call(this, inputNodes)
    at EslintValidationFilter.Plugin.__broccoliGetInfo__ (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/broccoli-plugin/index.js:55:43)
    at new ReadCompat (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/broccoli-plugin/read_compat.js:15:33)
    at EslintValidationFilter.Plugin._initializeReadCompat (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/broccoli-plugin/index.js:130:22)
    at EslintValidationFilter.Plugin.read (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/broccoli-plugin/index.js:110:12)
    at /Users/devinus/Projects/broccoli-lint-eslint/node_modules/broccoli/lib/builder.js:106:23
    at lib$rsvp$$internal$$tryCatch (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/rsvp/dist/rsvp.js:493:16)
    at lib$rsvp$$internal$$invokeCallback (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/rsvp/dist/rsvp.js:505:17)
    at /Users/devinus/Projects/broccoli-lint-eslint/node_modules/rsvp/dist/rsvp.js:1001:13
    at lib$rsvp$asap$$flush (/Users/devinus/Projects/broccoli-lint-eslint/node_modules/rsvp/dist/rsvp.js:1198:9)
    at doNTCallback0 (node.js:430:9)

Build failed
npm ERR! Test failed.  See above for more details.
```

--- a/.eslintrc
+++ b/.eslintrc
@@ -11,6 +11,9 @@
     "space-after-keywords": 0,
     "space-before-keywords": 0,
     "space-return-throw-case": 0,
-    "keyword-spacing": 2
+    "keyword-spacing": 2,
+    "newline-per-chained-call": 0,
+    "no-magic-numbers": 0,
+    "arrow-spacing": 0
   }
 }

--- a/.eslintrc
+++ b/.eslintrc
@@ -4,6 +4,13 @@
     "no-console": 0,
     "prefer-arrow-callback": 0,
     "prefer-reflect": 0,
-    "prefer-template": 0
+    "prefer-template": 0,
+    "no-restricted-syntax": [2, "WithStatement"],
+    "no-empty-label": 0,
+    "no-labels": 2,
+    "space-after-keywords": 0,
+    "space-before-keywords": 0,
+    "space-return-throw-case": 0,
+    "keyword-spacing": 2
   }
 }

--- a/Brocfile.js
+++ b/Brocfile.js
@@ -1,3 +1,5 @@
+'use strict';
+
 var eslint = require('./lib/index');
 var mergeTrees = require('broccoli-merge-trees');
 

--- a/Brocfile.js
+++ b/Brocfile.js
@@ -1,17 +1,15 @@
-'use strict';
-
-var eslint = require('./lib/index');
-var mergeTrees = require('broccoli-merge-trees');
+const eslint = require('./lib/index');
+const mergeTrees = require('broccoli-merge-trees');
 
 // lint plugin code
-var plugin = eslint('lib', {
+const plugin = eslint('lib', {
   options: {
     ignore: false
   }
 });
 
 // lint tests
-var test = eslint('test', {
+const test = eslint('test', {
   options: {
     ignore: false,
     rulePaths: ['conf/rules'],

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+# develop
+- Update to eslint@2.2.0
+
 # 1.1.1
 - Update to eslint@1.4.1
 

--- a/conf/rules/no-alert.js
+++ b/conf/rules/no-alert.js
@@ -1,5 +1,3 @@
-'use strict';
-
 module.exports = function noAlerts(context) {
   return {
     CallExpression: function callExpression(node) {
@@ -9,4 +7,3 @@ module.exports = function noAlerts(context) {
     }
   };
 };
-

--- a/conf/rules/no-alert.js
+++ b/conf/rules/no-alert.js
@@ -1,3 +1,5 @@
+'use strict';
+
 module.exports = function noAlerts(context) {
   return {
     CallExpression: function callExpression(node) {

--- a/lib/index.js
+++ b/lib/index.js
@@ -1,3 +1,7 @@
+/* eslint global-require: 0, consistent-return: 0 */
+
+'use strict';
+
 var Filter = require('broccoli-filter');
 var CLIEngine = require('eslint').CLIEngine;
 

--- a/lib/index.js
+++ b/lib/index.js
@@ -1,9 +1,7 @@
 /* eslint global-require: 0, consistent-return: 0 */
 
-'use strict';
-
-var Filter = require('broccoli-filter');
-var CLIEngine = require('eslint').CLIEngine;
+const Filter = require('broccoli-filter');
+const CLIEngine = require('eslint').CLIEngine;
 
 /**
  * Calculates the severity of a eslint.linter.verify result
@@ -34,7 +32,7 @@ function getResultSeverity(result) {
  * @returns {Array} filtered errors
  */
 function filterIgnoredFileMessages(errors) {
-  var ignoreRegex = /File ignored because of your \.eslintignore file/;
+  const ignoreRegex = /File ignored because of your \.eslintignore file/;
 
   return errors.filter(function filterErrors(error) {
     if (error.message.match(ignoreRegex)) {
@@ -50,7 +48,7 @@ function filterIgnoredFileMessages(errors) {
  * @returns {Object} filtered results
  */
 function filterAllIgnoredFileMessages(result) {
-  var resultOutput = result;
+  const resultOutput = result;
 
   result.results.map(function filterIgnoredErrors(resultItem) {
     resultItem.messages = filterIgnoredFileMessages(resultItem.messages);
@@ -93,21 +91,19 @@ EslintValidationFilter.prototype.extensions = ['js'];
 EslintValidationFilter.prototype.targetExtension = 'js';
 
 EslintValidationFilter.prototype.write = function write(readTree, destDir) {
-  var that = this;
-
-  return readTree(this.inputTree).then(function writeTree(srcDir) {
-    if (!that.eslintrc) {
-      that.eslintrc = srcDir;
+  return readTree(this.inputTree).then(srcDir => {
+    if (!this.eslintrc) {
+      this.eslintrc = srcDir;
     }
 
-    return Filter.prototype.write.call(that, readTree, destDir);
+    return Filter.prototype.write.call(this, readTree, destDir);
   });
 };
 
 EslintValidationFilter.prototype.processString = function processString(content, relativePath) {
   // verify file content
-  var result = this.cli.executeOnFiles([this.eslintrc + '/' + relativePath]);
-  var filteredResults = filterAllIgnoredFileMessages(result);
+  const result = this.cli.executeOnFiles([this.eslintrc + '/' + relativePath]);
+  const filteredResults = filterAllIgnoredFileMessages(result);
 
   // if verification has result
   if (filteredResults.results[0].messages.length) {

--- a/package.json
+++ b/package.json
@@ -35,7 +35,7 @@
     "broccoli": "^0.16.9",
     "broccoli-cli": "^1.0.0",
     "broccoli-merge-trees": "^1.1.1",
-    "eslint-config-nightmare-mode": "^0.2.0",
+    "eslint-config-nightmare-mode": "^2.2.0",
     "mocha": "^2.2.4",
     "rimraf": "^2.3.2"
   }

--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
   "main": "lib/index.js",
   "scripts": {
     "lint": "eslint .",
-    "test": "rm -rf build/ && npm run lint && broccoli build build > broccoli-build.out && node node_modules/mocha/bin/mocha build/test.js"
+    "test": "rm -rf build/ && npm run lint && ./node_modules/broccoli-cli/bin/broccoli build build > broccoli-build.out && node node_modules/mocha/bin/mocha build/test.js"
   },
   "repository": {
     "type": "git",
@@ -29,11 +29,12 @@
   "homepage": "https://github.com/jonathanKingston/broccoli-lint-eslint",
   "dependencies": {
     "broccoli-filter": "^0.1.12",
-    "eslint": "^1.4.1"
+    "eslint": "^2.2.0"
   },
   "devDependencies": {
-    "broccoli": "^0.16.2",
-    "broccoli-merge-trees": "^0.2.1",
+    "broccoli": "^0.16.9",
+    "broccoli-cli": "^1.0.0",
+    "broccoli-merge-trees": "^1.1.1",
     "eslint-config-nightmare-mode": "^0.2.0",
     "mocha": "^2.2.4",
     "rimraf": "^2.3.2"

--- a/test/.eslintrc
+++ b/test/.eslintrc
@@ -3,13 +3,18 @@
   "rules": {
     "no-console": 0,
     "prefer-arrow-callback": 0,
+    "prefer-reflect": 0,
+    "prefer-template": 0,
+    "no-restricted-syntax": [2, "WithStatement"],
     "no-empty-label": 0,
     "no-labels": 2,
-    "no-restricted-syntax": [2, "WithStatement"],
     "space-after-keywords": 0,
     "space-before-keywords": 0,
     "space-return-throw-case": 0,
-    "keyword-spacing": 2
+    "keyword-spacing": 2,
+    "newline-per-chained-call": 0,
+    "no-magic-numbers": 0,
+    "arrow-spacing": 0
   },
   "env": {
     "mocha": true

--- a/test/.eslintrc
+++ b/test/.eslintrc
@@ -2,7 +2,14 @@
   "extends": "nightmare-mode/node",
   "rules": {
     "no-console": 0,
-    "prefer-arrow-callback": 0
+    "prefer-arrow-callback": 0,
+    "no-empty-label": 0,
+    "no-labels": 2,
+    "no-restricted-syntax": [2, "WithStatement"],
+    "space-after-keywords": 0,
+    "space-before-keywords": 0,
+    "space-return-throw-case": 0,
+    "keyword-spacing": 2
   },
   "env": {
     "mocha": true

--- a/test/test.js
+++ b/test/test.js
@@ -1,7 +1,6 @@
-'use strict';
-var assert = require('assert');
-var fs = require('fs');
-var rimraf = require('rimraf');
+const assert = require('assert');
+const fs = require('fs');
+const rimraf = require('rimraf');
 
 afterEach(function afterEach() {
   rimraf.sync('temp');
@@ -9,11 +8,11 @@ afterEach(function afterEach() {
 });
 
 it('should reported errors', function shouldReportErrors() {
-  var buildLog = fs.readFileSync('broccoli-build.out').toString();
+  const buildLog = fs.readFileSync('broccoli-build.out').toString();
+  const NOT_FOUND = -1;
 
-  assert(buildLog.indexOf('Strings must use doublequote.') !== -1, 'Used eslint validation - strings');
-  assert(buildLog.indexOf('is not in camel case') !== -1, 'Used eslint validation - camel case');
-  assert(buildLog.indexOf('testing custom rules') !== -1, 'Used custom rulesdir rules');
-  assert(buildLog.indexOf('fixture/1.js') !== -1, 'Shows filepath');
+  assert(buildLog.indexOf('Strings must use doublequote.') !== NOT_FOUND, 'Used eslint validation - strings');
+  assert(buildLog.indexOf('is not in camel case') !== NOT_FOUND, 'Used eslint validation - camel case');
+  assert(buildLog.indexOf('testing custom rules') !== NOT_FOUND, 'Used custom rulesdir rules');
+  assert(buildLog.indexOf('fixture/1.js') !== NOT_FOUND, 'Shows filepath');
 });
-

--- a/test/test.js
+++ b/test/test.js
@@ -1,3 +1,4 @@
+'use strict';
 var assert = require('assert');
 var fs = require('fs');
 var rimraf = require('rimraf');

